src/Encode.elm

```diff
@@ -12,76 +12,193 @@ toBytes string =
 
 encoder : String -> Maybe Encode.Encoder
 encoder string =
-    string
-        |> stringToInts
-        |> Maybe.map (List.map Encode.unsignedInt8)
-        |> Maybe.map Encode.sequence
+    encodeChunks string []
+        |> Maybe.map (List.reverse >> Encode.sequence)
 
 
-stringToInts : String -> Maybe (List Int)
-stringToInts string =
-    string
-        |> String.toList
-        |> quadMap fourCharsToThreeInts
-        |> listFromMaybeList
-        |> Maybe.map List.concat
+{-| Big picture:
 
+  - read 4 base64 characters
+  - convert them to 3 bytes (24 bits)
+  - encode these bytes
 
-fourCharsToThreeInts : Char -> Maybe Char -> Maybe Char -> Maybe Char -> Maybe (List Int)
-fourCharsToThreeInts a b c d =
-    case fourCharsToInt24 a b c d of
-        Nothing ->
+@TODO this means the input length must be a multiple of 4. Is that too strict?
```

Comment:
I think this could be solved by modifying `encoder`:

```
encoder string =
    case modBy 4 (String.length string) of
        0 ->
            encodeChunks string []
                |> Maybe.map (List.reverse >> Encode.sequence)

        2 ->
            encoder (string ++ "==")

        3 ->
            encoder (string ++ "=")

        _ ->
            Nothing
```

What do you think?

Reply:
I put it in encodeChunks. These cases will happen rarely, but now we don't pay any cost at all if it doesn't happen. Also added tests to make sure that omitted characters are treated the same as padding.